staging/src/k8s.io/client-go/metadata/fake/simple.go

```diff
@@ -0,0 +1,390 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fake
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/metadata"
+	"k8s.io/client-go/testing"
+)
+
+// MetadataClient assists in creating fake objects for use when testing, since metadata.Getter
+// does not expose create
+type MetadataClient interface {
+	metadata.Getter
+	Create(obj *metav1.PartialObjectMetadata, opts metav1.CreateOptions, subresources ...string) (*metav1.PartialObjectMetadata, error)
```

Comment:
Would it be a good idea to make the method name clearly for testing only? Eg CreateFake?

Reply:
It’s not really any different than the other methods (patch, etc)

Reply:
It seems like we should not be trying to create anything via the metadata client. So having the fake able to do so seems like it could mask an issue where the real client can't, but the test makes it seems like this works.

Reply:
If I'm someone looking at the fake directly, Create or CreateFake has the same impl.  If we ever add Create to metadata client (or someone using dynamic switches tests to use the metadata client, which is the use case here), Create is more consistent with our existing client logic than CreateFake.

Anyone who is using the Fake client inside of production code isn't going to work anyway, so I don't see how the fake client could cause that outcome.  I'd bias towards not creating a new pattern (CreateFake) in this case to remain consistent with all other clients.

Reply:
Jordan also commented on this, if two people tell me something independently they're probably right.

Updated :)